packages/block-library/src/site-logo/index.php

```diff
@@ -72,4 +91,63 @@ function register_block_core_site_logo() {
 		)
 	);
 }
+
 add_action( 'init', 'register_block_core_site_logo' );
+
+/**
+ * Overrides the custom logo with a site logo, if the option is set.
+ *
+ * @param string $custom_logo The custom logo set by a theme.
+ *
+ * @return string The site logo if set.
+ */
+function _override_custom_logo_theme_mod( $custom_logo ) {
+	$site_logo = get_option( 'site_logo' );
+	return false === $site_logo ? $custom_logo : $site_logo;
+}
+
+add_filter( 'theme_mod_custom_logo', '_override_custom_logo_theme_mod' );
+
+/**
+ * Updates the site_logo option when the custom_logo theme-mod gets updated.
+ *
+ * @param string $custom_logo The custom logo set by a theme.
+ *
+ * @return string The custom logo.
+ */
+function _sync_custom_logo_to_site_logo( $custom_logo ) {
+	// Delete the option when the custom logo does not exist or was removed.
+	// This step ensures the option stays in sync.
+	if ( empty( $custom_logo ) ) {
+		delete_option( 'site_logo' );
+	} else {
+		remove_action( 'update_option_site_logo', '_sync_site_logo_to_custom_logo' );
+		update_option( 'site_logo', $custom_logo );
+		add_action( 'update_option_site_logo', '_sync_site_logo_to_custom_logo', 10, 2 );
+	}
+	return $custom_logo;
+}
+
+add_filter( 'pre_set_theme_mod_custom_logo', '_sync_custom_logo_to_site_logo' );
+
+/**
+ * Updates the custom_logo theme-mod when the site_logo option gets updated.
+ *
+ * @param mixed $old_value The old option value.
+ * @param mixed $value     The new option value.
+ *
+ * @return void
+ */
+function _sync_site_logo_to_custom_logo( $old_value, $value ) {
+	// Delete the option when the custom logo does not exist or was removed.
+	// This step ensures the option stays in sync.
+	if ( empty( $value ) ) {
+		remove_theme_mod( 'custom_logo' );
+	} else {
+		remove_filter( 'pre_set_theme_mod_custom_logo', '_sync_custom_logo_to_site_logo' );
+		set_theme_mod( 'custom_logo', $value );
+		add_filter( 'pre_set_theme_mod_custom_logo', '_sync_custom_logo_to_site_logo' );
+	}
+}
+
+add_action( 'update_option_site_logo', '_sync_site_logo_to_custom_logo', 10, 2 );
```

Comment:
We were fixing an issue last night where this code sync'd values from a conflicting site option in Jetpack. 

One possible solution might be to check that the current theme supports the `custom_logo` theme mod, and if not, then we don't register this action.

/cc @creativecoder @kraftbj @fullofcaffeine